src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -741,5 +771,79 @@ private void UpdateCacheExpirationTime(KeyValueWatcher changeWatcher, bool succe
 
             changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(cacheExpirationTime);
         }
+
+        private async Task ExecuteWithFailOverPolicyAsync(Func<ConfigurationClient, Task> funcToExecute, CancellationToken cancellationToken = default)
+        {
+            IEnumerator<ConfigurationClient> clientEnumerator = _configClientProvider.GetClientEnumerator();
+
+            // Guaranteed to return at least 1 element
+            Debug.Assert(clientEnumerator.MoveNext());
+
+            while (true)
+            {
+                try
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await funcToExecute(clientEnumerator.Current);
+
+                    _configClientProvider.UpdateClientStatus(clientEnumerator.Current, successful: true);
+
+                    break;
+                }
+                catch (AggregateException ae)
+                {
+                    if (!ShouldFailoverForException(ae))
+                    {
+                        throw;
+                    }
+                    else
+                    {
+                        _configClientProvider.UpdateClientStatus(clientEnumerator.Current, successful: false);
+
+                        if (!clientEnumerator.MoveNext())
+                        {
+                            throw;
+                        }
+                    }
+                }
+                catch (RequestFailedException e)
+                {
+                    if (!ShouldFailoverForException(e))
+                    {
+                        throw;
+                    }
+                    else
+                    {
+                        _configClientProvider.UpdateClientStatus(clientEnumerator.Current, successful: false);
+
+                        if (!clientEnumerator.MoveNext())
+                        {
+                            throw;
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool ShouldFailoverForException(Exception ex)

```

Comment:
What about socket exception? and other retriable status codes like 503? Is this incomplete or is it being handled elsewhere?

Reply:
`SocketException` is wrapped under `RequestFailedException` so that is covered under the `else if` condition.
Currently we retry on the following status codes - 
1. 429 - Too Many Requests
2. 408 - Request Timeout
3. Status >= 500  (Status code 503 comes under this condition)

Are there other status codes you think we should retry on?

Reply:
But socket exception would not have a status code defined. So doesn't look like you'll be considering it to be failover-able?

Reply:
socket exception doesn't have a status code, I'm having trouble finding how it's being handled as failover-able.

Reply:
I see what you are saying. I updated the code now to account for SocketException and WebException.